syntax = "proto3";

package pb;

message SyncPid
{
    int32 Pid = 1;
    string UserName = 2;
}

message Player
{
    int32 Pid = 1;
    Position P = 2;
    string UserName = 3;
}

message SyncPlayers
{
    repeated Player ps = 1;
}

message Position{
    float X = 1;
    float Y = 2;
    float Z = 3;
    float V = 4;
    int32 BloodValue = 5;
}

message MovePackage
{
    int32 Pid = 1;
    int32 ActionData = 2;
}

message BroadCast
{
    int32 Pid = 1;
    int32 Tp = 2;
    oneof Data 
    {
        string Content = 3;
        Position P = 4;
        /* ActionData 暂时预留 */
        int32 ActionData = 5;
    }
    string UserName = 6;
}

message Talk
{
    string Contect = 1;
}